algorithms/color_refinement.py

```diff
@@ -36,30 +41,38 @@ def __initialize_colors(G: "Graph"):
     return colors, max_colornum
 
 
-def __colorgroup_refinement(colors, vertices: List["Vertex"], max_colornum):
+def __colorgroup_refinement(colors, colornum, vertices: List["Vertex"], next_colornum):
     """
-
-    :param vertices:
-    :param max_colornum:
-    :return:
+    For vertices with property colornum = 'colornum', define which vertices have equal neighbours and which do not
+    If vertices within this group have not the same neighbours, these vertices will be given a new colornum and put in
+    a new dictionary key
+    :param colors: Dictionary with property colornum as key and a list of vertices with that property as value
+    :param colornum: Current colornum to be evaluated
+    :param vertices: Vertices of that colornum to be evaluated
+    :param next_colornum: If vertices are different, give those vertices the next_colornum as property colornum
     """
+
+    # Creates a list of alle colornums of the neighbours of the first vertex in 'vertices'
     v0 = vertices[0]
     v0_colors = []
     for v in v0.neighbours:
         v0_colors.append(v.colornum)
-    has_changed = False
+
+    # Compare the neighbours of each vertex in 'vertices' with the first vertex in 'vertices'
```

Comment:
Eigenlijk vergelijken we hier iedere vertex in 'vertices' met de eerste in 'vertices'.